docs/cep/windows/window-types/sliding-expression.md

```diff
@@ -0,0 +1,29 @@
+---
+title: SLIDING_EXPRESSION()
+---
+
+A sliding window that holds the last number of events defined by an expression at a given time, and gets updated for each arrival and expiration.
+
+## Syntax
+
+    WINDOW SLIDING_EXPRESSION(expression <STR>)
+
+## Query Parameters
+
+| Name          | Description     | Default Value | Possible Data Types | Optional | Dynamic |
+|----------|---------------------------------------|-------------|-----------------|----------|---------|
+| expression | An expression that defines what events should be included in a sliding length window. All possible expressions are allowed. |          | STR        | No       | No      |
+
+## Example
+
+```sql
+CREATE STREAM cseEventStream (symbol string, price float, volume int);
```

Comment:
This query is invalid because I do not know what expression to enter in the window that will work. @StoyanTc and @koshy-thomas, please help.

Also, please give this whole topic a thorough read, as it is brand new.

Reply:
@diana-macrometa 
You can use this SW as example:

CREATE STREAM cseEventStream (symbol string, price float, volume int);
CREATE WINDOW cseEventWindow (symbol string, price float, volume int) SLIDING_EXPRESSION('count() <= 20');
CREATE SINK OutputStream WITH (type='stream', stream='OutputStream') (symbol string, price double);

@info(name = 'query0')
INSERT INTO cseEventWindow
FROM cseEventStream;

@info(name = 'query1')
INSERT INTO OutputStream
SELECT symbol, sum(price) AS price
FROM cseEventWindow;

As for the content, it says you can give anything there.. so that part probably needs to be rephrased.. since surely some kind of valid expression is needed I believe but I am not aware of exact constraints so atleast we should avoid saying anything can be given as expression here??

Reply:
That's what @StoyanTc told me.